Remotion/Validation/UnitTests/Validators/ScalePrecisionValidatorTest.cs

```diff
@@ -0,0 +1,171 @@
+      var validator = new ScalePrecisionValidator (2, 4);
+
+      var validationFailures = validator.Validate (propertyValidatorContext);
+
+      Assert.That (validationFailures, Is.Empty);
+    }
+
+    [Test]
+    public void Validate_WithPropertyValueScaleGreaterThanValidatorValue_ReturnsSingleValidationFailure ()
+    {
+      var propertyValidatorContext = CreatePropertyValidatorContext (10.123m);
+      var validator = new ScalePrecisionValidator (2, 5);
+
+      var validationFailures = validator.Validate (propertyValidatorContext).ToArray();
+
+      Assert.That (validationFailures.Length, Is.EqualTo (1));
+      Assert.That (
+          validationFailures[0].ErrorMessage,
+          Is.EqualTo ("!!! The value must not be more than 5 digits in total, with allowance for 2 decimals. 2 digits and 3 decimals were found."));
+      Assert.That (
+          validationFailures[0].LocalizedValidationMessage,
+          Is.EqualTo ("!!! The value must not be more than 5 digits in total, with allowance for 2 decimals. 2 digits and 3 decimals were found."));
+    }
+
+    [Test]
+    public void Validate_WithPropertyValuePrecisionGreaterThanValidatorValue_ReturnsSingleValidationFailure ()
+    {
+      var propertyValidatorContext = CreatePropertyValidatorContext (10.123m);
+      var validator = new ScalePrecisionValidator (3, 4);
+
+      var validationFailures = validator.Validate (propertyValidatorContext).ToArray();
+
+      Assert.That (validationFailures.Length, Is.EqualTo (1));
+      Assert.That (
+          validationFailures[0].ErrorMessage,
+          Is.EqualTo ("!!! The value must not be more than 4 digits in total, with allowance for 3 decimals. 2 digits and 3 decimals were found."));
+    }
+
+    [Test]
+    public void Validate_WithPropertyValuePrecisionGreaterThanValidatorValueAndCustomValidationMessage_ReturnsSingleValidationFailureWithCustomValidationMessage ()
+    {
+      var propertyValidatorContext = CreatePropertyValidatorContext (10.123m);
+      var validator = new ScalePrecisionValidator (
+          3,
+          4,
+          new InvariantValidationMessage (
+              "Custom validation message: '{ExpectedPrecision}', '{ExpectedScale}', '{Digits}', '{ActualScale}'."));
+
+      var validationFailures = validator.Validate (propertyValidatorContext).ToArray();
+
+      Assert.That (validationFailures.Length, Is.EqualTo (1));
+      Assert.That (
+          validationFailures[0].ErrorMessage,
+          Is.EqualTo ("!!! The value must not be more than 4 digits in total, with allowance for 3 decimals. 2 digits and 3 decimals were found."));
+      Assert.That (validationFailures[0].LocalizedValidationMessage, Is.EqualTo ("Custom validation message: '4', '3', '2', '3'."));
+    }
+
+    [Test]
+    public void Validate_WithPropertyValueScaleGreaterThanValidatorValueAndIgnoreTrailingZeros_ReturnsNoValidationFailures ()
+    {
+      var propertyValidatorContext = CreatePropertyValidatorContext (10.12300m);
```

Comment:
ah, okay. yeah wil lwant to try this wit hignoretrailigzeros = false and just 3 decimal places.